app-modules/service-management/src/Models/Contract.php

```diff
@@ -0,0 +1,96 @@
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Aiding App™ is licensed under the Elastic License 2.0. For more details,
+    see <https://github.com/canyongbs/aidingapp/blob/main/LICENSE.>
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AidingApp\ServiceManagement\Models;
+
+use Spatie\MediaLibrary\HasMedia;
+use Illuminate\Database\Eloquent\Model;
+use OwenIt\Auditing\Contracts\Auditable;
+use Spatie\MediaLibrary\InteractsWithMedia;
+use Illuminate\Database\Eloquent\Concerns\HasUuids;
+use AidingApp\ServiceManagement\Enums\ContractStatus;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use AidingApp\ServiceManagement\Casts\ContractCurrencyValue;
+use AidingApp\Audit\Models\Concerns\Auditable as AuditableTrait;
+
+/**
+ * @mixin IdeHelperContract
+ */
+class Contract extends Model implements HasMedia, Auditable
+{
+    use HasFactory;
+    use HasUuids;
+    use InteractsWithMedia;
+    use AuditableTrait;
+
+    protected $append = ['status'];
+
+    protected $fillable = [
+        'name',
+        'description',
+        'contract_type',
+        'vendor_name',
+        'start_date',
+        'end_date',
+        'contract_value',
+    ];
+
+    protected $casts = [
+        'contract_value' => ContractCurrencyValue::class,
+    ];
+
+    public function contractType(): BelongsTo
+    {
+        return $this->belongsTo(ContractType::class, 'contract_type', 'id');
+    }
+
+    public function getStatusAttribute(): ContractStatus
+    {
+        return ContractStatus::getStatus($this->start_date, $this->end_date);
+    }
```

Comment:
This is not how an accessor is traditionally written in Laravel. Let's make sure we use the new standard way of doing so: https://laravel.com/docs/10.x/eloquent-mutators#defining-an-accessor